public/js/index.js

```diff
@@ -2375,14 +2375,42 @@ function initTopicbar() {
         },
     });
 }
-function toggleDuedateForm() {
-    $('#add_deadline_form').fadeToggle(150);
+function toggleDeadlineForm() {
+    $('#deadlineForm').fadeToggle(150);
 }
 
-function deleteDueDate(url) {
-    $.post(url, {
-        '_csrf': csrf,
-    },function( data ) {
-        window.location.reload();
+function setDeadline() {
+    var deadline = $('#deadlineDate').val();
+    updateDeadline(deadline);
+}
+
+function updateDeadline(deadlineString) {
+    $('#deadline-err-invalid-date').hide();
+    $('#deadline-loader').addClass('loading');
+
+    var realDeadline = null;
+    if (deadlineString !== '') {
+        var dateToMatch = /([0-9]){4}-([0-9]){2}-([0-9]){2}\b/
```

Comment:
Please use `Date.parse(deadlineString)` instead. If the result is `NaN` it is an invalid date. If the result is valid, you can use `new Date(result)` directly in your JSON response. It will be encoded as `2012-04-23T18:25:43.511Z`.